presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -389,6 +390,10 @@ else if (actualReturnType.isPrimitive()) {
 
             this.specializedTypeParameters = getDeclaredSpecializedTypeParameters(method);
 
+            for (TypeParameter typeParameter : typeParameters) {
+                checkArgument(typeParameter.value().matches("[A-Z]+"), "Type parameter %s annotated on method [%s] should only contain A-Z", typeParameter.value(), method);
```

Comment:
Write message like `Expected type parameter to only contain A-Z, but got %s on method [%s].`